From: SeongJae Park <sj@kernel.org>
Date: Sat, 14 Jun 2025 14:05:20 -0700
Subject: [PATCH] ==== lru_sort modernization ====

mm/damon: advance DAMON-based LRU sorting

DAMOS can be used for more access-aware LRU sorting, using
DAMOS_LRU_[DE]PRIO actions.  DAMON_LRU_SORT is a kernel module for
utilizing DAMOS for the specific use case in simpler ways.

Modern DAMOS features including page level filtering, quotas auto-tuning
and monitoring intervsals auto-tuning can help the access-aware LRU
sorting.

/** !!! TODO: More wording **/

Patches Sequence
================

First four patches extend DAMOS quota auto-tuning to support [in]active
memory ratio target metric type.  The patches define new metrics
(patch 1), implement core layer support (patch 2), DAMON sysfs support
(patch 3), and add the document (patch 4).

Following patch (patch 5) make DAMOS_LRU_PRIO action to directly move
target pages to active LRU list, instead of marking accessed and
incrementally be moved to active lists.

Following seven patches (patches 6-12) updates DAMON_LRU_SORT to support
modern DAMON features.  Patch 6 makes it to use not only access
frequency but also age at under-quota regions prioritization.  Patch
7-12 add the support and document it for young page filtering,
active:inactive memory ratio based quota auto-tuning, and monitoring
intervals auto-tuning, respectively.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_mv9z21za | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_mv9z21za

diff --git a/hkml_cv_bogus/hkml_cv_bogus_mv9z21za b/hkml_cv_bogus/hkml_cv_bogus_mv9z21za
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

